src/scss/components/carousel.scss

```diff
@@ -38,50 +33,37 @@
       h2,
       h3,
       p,
-      a {
+      a:not(.btn) {
         color: $carousel-caption-color;
         // Add text shadow to all text if there is an image
         text-shadow: 1px 1px 1px rgba(0, 0, 0, 0.5);
       }
-
-      .btn {
-        text-shadow: none;
-      }
-
-      .btn-link {
-        color: $carousel-caption-color;
-        border-color: $carousel-control-color;
-
-        &:hover {
-          border-color: darken($carousel-control-color, 20%);
-        }
-      }
     }
 
     .btn {
       margin-left: $line-height-computed * 0.45;
       margin-right: $line-height-computed * 0.45;
-      padding: ($line-height-computed * 0.5) ($line-height-computed * 0.7);
-      @include theme-contrast-color("accent", "color");
 
       @include media-breakpoint-down(sm) {
         margin-bottom: floor($line-height-computed * 0.45);
       }
     }
 
-    .btn-link {
-      position: relative;
-      border: 1px solid lighten($primary, 20%);
-      @include theme-color("accent", "border-color");
+    &:not(.carousel-has-image) .btn-secondary-action {
       @include theme-contrast-color("carousel", "color");
-      transition: all 0.3s ease;
+      border-color: currentColor;
 
       &:hover {
-        border-color: lighten($primary, 40%);
-        @include theme-color("accent", "border-color", "lighten", 20%);
+        @include theme-contrast-color("carousel", "color", #000);
       }
     }
   }
+
+  .theme-carousel-default-background & {
+    a:not(.btn) {
```

Comment:
This could be combined into `.theme-carousel-default-background & a:not(.btn) {`.